### PR TITLE
Add method to extract subvariables from variable tuples

### LINKED ISSUE #215: Enrich the variable catalog generated by CrunchCube objects
`variables()` on a CrunchCube return a variable catalog with variable tuples which don't have all the same methods for accessing subvariables. 

We should work on making methods like `subvariables()`, `names()`, `aliases()` etc. work on variable tuples (when that information is available, eg variable tuples form CrunchCubes had subreferences with names and aliases, however variable tuples from the variable catalog doesn't).

- [x] `names()` for subvariables/subreferneces from a CrunchCube
- [x] `aliases()` for subvariables/subreferneces from a CrunchCube

### PR DESCRIPTION
**Pull request by jonkeane**
Closes #215

### CODE REVIEW

File: R/subvariables.R [hunk 1]
```diff
@@ -34,27 +34,43 @@ setMethod("subvariables", "CategoricalArrayVariable", function (x) {
...
+            # if there is a subvariable element, we need to use that for ordering
+            names(subvars) <- paste0(absoluteURL(names(subvars), base=catalog_url), "/")
+            out <- Subvariables(
+                index=subvars[x$subvariables],
+                self=catalog_url
+            )
+        } else {
+            # if there is no subvariable element, this might be from a
+            # CrunchCube which doesn't have a subvariables member, though the
+            # order *might* not be deterministic, checking zz9 source still.
```
Comment (nealrichardson) [inline R/subvariables.R]: You could inject a "subvariables" member in `cubeVarReferences()` based on the dimension.

File: R/cube-dims.R [hunk 2]
```diff
@@ -39,6 +39,25 @@ cubeVarReferences <- function (x) {
...
+                                     function (x) x$alias %||% "",
+                                     character(1))
+
+        if (is.null(names(tuple$subreferences))) {
+            # if there are no names for the subvariable elements, fake urls from
+            # the aliases
+            names(tuple$subreferences) <- tuple$subvariables
+        }
+
+        # add a trailling slash to match how urls will look.
```
Comment (nealrichardson) [inline R/cube-dims.R]: Why is this necessary?

File: R/subvariables.R [hunk 3]
```diff
@@ -34,27 +34,29 @@ setMethod("subvariables", "CategoricalArrayVariable", function (x) {
...
+    return(absoluteURL(unlist(x$subvariables), base=x@index_url))
+}
+
 #' @rdname Subvariables
 #' @export
 setMethod("subvariables", "VariableTuple", function (x) {
-    ## Return subvariable *urls* from a Tuple, properly formatted and absolute
-    return(absoluteURL(unlist(x$subvariables), base=x@index_url))
+    catalog_url <- absoluteURL(x$subvariables_catalog, base=x@index_url) %||% ""
+        subvars <- x$subreferences
```
Comment (nealrichardson) [inline R/subvariables.R]: Is it just GitHub, or is the indentation inconsistent here?

Comment (jonkeane) [reply R/subvariables.R]: You're right, the indentation was wrong here. I've fixed it.

File: R/cube-dims.R (hunk 2 above)
Comment (jonkeane) [reply R/cube-dims.R]: The subvariables from a variable entity end with a `/` since they are full URLS (https://github.com/Crunch-io/rcrunch/blob/master/inst/app.crunch.io/api/datasets/1/variables/mymrset.json#L20-L24) 

With cube responses, here, we are faking the url from just the variable id, which lacks the trailing `/` (https://github.com/Crunch-io/rcrunch/blob/master/tests/testthat/cubes/mr-by-mr-different-mrs.json#L231-L234).

This allows the subvariable method for variable tuples to be more general and not special case cubes (at this point specifically: https://github.com/Crunch-io/rcrunch/pull/242/files#diff-d2a62bc0f07894f773e0ab8ecd51becaR57)
